core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,497 @@
+          });
+
+      const questionContributionSlicedStats: Stats[] = this.sliceStatsIntoPage(
+        questionContributionStats);
+
+      this.questionContributionPagedStats.push(
+        this.generatePagedStats(questionContributionSlicedStats));
+
+      this.statsData.questionContribution = this.questionContributionPagedStats;
+    }
+
+    if (response.question_review_stats.length > 0) {
+      const questionReviewStats: QuestionReviewStats[] = response.
+        question_review_stats.map((stat) => {
+          return this.createQuestionReviewStat(stat);
+        });
+
+      const questionReviewSlicedStats: Stats[] = this.sliceStatsIntoPage(
+        questionReviewStats);
+
+      this.questionReviewPagedStats.push(
+        this.generatePagedStats(questionReviewSlicedStats));
+
+      this.statsData.questionReview = this.questionReviewPagedStats;
+    }
+  }
+
+  fetchLanguagesFromStats(
+      stats:
+        TranslationContributionBackendDict[] |
+        TranslationReviewBackendDict[]): Set<string> {
+    const languages = new Set<string>();
+    stats.map((stat) => {
+      languages.add(stat.language_code);
+    });
+
+    return languages;
+  }
+
+  createTranslationContributionStat(
+      stat: TranslationContributionBackendDict): TranslationContributionStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      acceptedCards: stat.accepted_translations_count,
+      acceptedWordCount: stat.accepted_translation_word_count
+    };
+  }
+
+  createTranslationReviewStat(
+      stat: TranslationReviewBackendDict): TranslationReviewStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      acceptedCards: stat.accepted_translations_count,
+      acceptedWordCount: stat.accepted_translation_word_count,
+      reviewedCards: stat.reviewed_translations_count,
+      reviewedWordCount: stat.reviewed_translation_word_count
+    };
+  }
+
+  createQuestionContributionStat(
+      stat: QuestionContributionBackendDict): QuestionContributionStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      acceptedQuestions: stat.accepted_questions_count,
+      acceptedQuestionsWithoutEdits: (
+        stat.accepted_questions_without_reviewer_edits_count)
+    };
+  }
+
+  createQuestionReviewStat(
+      stat: QuestionReviewBackendDict): QuestionReviewStats {
+    return {
+      firstContributionDate: stat.first_contribution_date,
+      lastContributionDate: stat.last_contribution_date,
+      topicName: stat.topic_name,
+      reviewedQuestions: stat.reviewed_questions_count,
+      acceptedQuestions: stat.accepted_questions_count
+    };
+  }
+
+  sliceStatsIntoPage(
```

Comment:
sliceStatsIntoPages (there are multiple pages, right)?

Reply:
Removed this function since it is no more used.